elasticgraph-graphql/spec/unit/elastic_graph/graphql/datastore_query/filtering_spec.rb

```diff
@@ -372,10 +476,98 @@ class GraphQL
             ]}
           })
 
-          expect(datastore_body_of(query)).to query_datastore_with({bool: {filter: [
-            {terms: {"tags" => ["a", "b"]}},
-            {terms: {"tags" => ["c", "d"]}}
-          ]}})
+          expect(datastore_body_of(query)).to query_datastore_with({bool: {
+            filter: [
+              {bool: {filter: [{terms: {"tags" => ["a", "b"]}}]}},
+              {bool: {filter: [{terms: {"tags" => ["c", "d"]}}]}}
+            ]
+          }})
+        end
+
+        it "handles not operator" do
```

Comment:
```suggestion
        it "handles `not` within `all_of` operator" do
```

...and then add a test covering `not: {all_of: ... }` please :).

Reply:
Done.